featureflag/featureflag.go

```diff
@@ -0,0 +1,66 @@
+package featureflag
+
+import (
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	ld "gopkg.in/launchdarkly/go-server-sdk.v4"
+)
+
+type Client interface {
+	Enabled(string, string) bool
+	Variation(string, string, string) string
+}
+
+type ldClient struct {
+	*ld.LDClient
+	log *logrus.Entry
+}
+
+func NewClientWithConfig(cfg *Config, logger *logrus.Entry) (Client, error) {
+	return NewClient(cfg.Key, cfg.RequestTimeout, logger)
+}
+
+func NewClient(key string, reqTimeout time.Duration, logger *logrus.Entry) (Client, error) {
```

Comment:
yeah, the intent here is to make the consumer code a bit cleaner. So instead of the consumer needing to know this is a `*featureflag.LDClient`, they can use the pointer-less `featureflag.Client`. `ScheduledExecutor` follows a similar pattern.

I don't know if there's huge gain in making it super flexible for future clients until we need to.

Reply:
I think this is fine for the time being.